streams/notification/src/main/java/com/hortonworks/streamline/streams/notification/store/hbase/mappers/StreamlineEventMapper.java

```diff
@@ -45,13 +45,14 @@
 
     @Override
     public StreamlineEvent entity(Result result) {
-        String id = Bytes.toString(result.getRow());
+        // FIXME: should we preserve event ID here? If then we should enable putting id to event which opens possibility of breaking from other parts.
+        //String id = Bytes.toString(result.getRow());
         Map<String, Object> fieldsAndValues = new HashMap<>();
         for(Map.Entry<byte[], byte[]> entry: result.getFamilyMap(CF_FIELDS).entrySet()) {
             fieldsAndValues.put(Bytes.toString(entry.getKey()), Bytes.toString(entry.getValue()));
         }
         String dataSourceId = Bytes.toString(result.getFamilyMap(CF_DATASOURCE_ID).firstEntry().getKey());
-        return new StreamlineEventImpl(fieldsAndValues, dataSourceId, id);
+        return StreamlineEventImpl.builder().fieldsAndValues(fieldsAndValues).dataSourceId(dataSourceId).build();
```

Comment:
Maybe Ok to return an event with a different ID than the one queried for now. Other option is to add a wrapper event with that takes and returns given "id" and delegates other calls to underlying event.